### PR TITLE
Upload/Download SBOM

### PR DESCRIPTION
**Pull request by eccles**
Problem:
Upload and download of SBOM is missing.

Solution:
Added code to upload and download SBOMS

Signed-off-by: Paul Hewlett <phewlett75@gmail.com>

### CODE REVIEW

File: archivist/sboms.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    def download(self, identity: str, fd: BinaryIO) -> Response:
+        """Read SBOM
+
+        Reads SBOM into data sink (usually a file opened for write)..
+        Note that returns the response as the body will be consumed by the
+        fd iterator
+
+        Args:
+            identity (str): SBOM identity e.g. sboms/xxxxxxxxxxxxxxxxxxxxxxx
+            fd (file): opened file escriptor or other file-type sink.
```
Comment (honourfish) [inline archivist/sboms.py:99]: minor: escriptor

File: unittests/testsboms.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+            )
+            self.assertEqual(
+                sbom.dict(),
+                RESPONSE,
+                msg="UPLOAD method called incorrectly",
+            )
+
+    def test_SBOMS_download(self):
+        """
+        Test attachment download
```
Comment (honourfish) [inline unittests/testsboms.py:133]: Test sbom download?

File: functests/execsboms.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+
+# pylint: disable=fixme
+# pylint: disable=missing-docstring
+# pylint: disable=unused-variable
+
+
+class TestSBOM(TestCase):
+    """
+    Test Archivist SBOM ipload/download
```
Comment (honourfish) [inline functests/execsboms.py:21]: minor: ipload

File: archivist/sboms.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        metadata: Optional[Dict] = None,
+    ):
+        """List SBOMS.
+
+        Lists SBOMS that match criteria:
+
+            sboms = self.arch.sboms.list(metadata={"uploaded_since": "2021-11-16T11:28:31Z"})
+
+        Args:
+            metadata (dict): optional e.g. {"life_cycle_status": "ACTIVE" }
```
Comment (honourfish) [inline archivist/sboms.py:148]: Minor: maybe we need a complete example of additional things we can add to the dict, stuff like `trusted`, `uploaded_since`, `uploaded_before`, `privacy`, `personal` with explanations for what each does?

File: archivist/sboms.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+from . import archivist as type_helper
+
+from .constants import SBOMS_SUBPATH, SBOMS_LABEL, SBOMS_METADATA, SBOMS_WILDCARD
+from .dictmerge import _deepmerge
+from .sbommetadata import SBOM
+
+#: Default page size - number of entities fetched in one REST GET in the
+#: :func:`~_AssetsClient.list` method. This can be overridden but should rarely
+#: be changed.
+DEFAULT_PAGE_SIZE = 500
```
Comment (honourfish) [inline archivist/sboms.py:45]: Minor: this needs to be synced up with the max page size which is tbc within the api, so we best keep an eye on both!

File: archivist/archivist.py [hunk 6]
```diff
@@ -346,7 +346,7 @@ def post_file(self, path: str, fd: BinaryIO, mtype: str) -> Dict:
 
         multipart = MultipartEncoder(
             fields={
-                "file": ("filename", fd, mtype),
+                form: ("filename", fd, mtype),
```
Comment (honourfish) [inline archivist/archivist.py:349]: nice!

**Review by honourfish — APPROVED** [review]

File: archivist/sboms.py (hunk 5 above)
Comment (eccles) [reply archivist/sboms.py:45]: I am going to leave this for now as it applies to other endpoints as well..
Correct solution is have a default of None

Comment (honourfish) [reply archivist/sboms.py:45]: sounds good!

Comment (eccles) [reply archivist/sboms.py:45]: Raised issue here https://github.com/jitsuin-inc/archivist-python/issues/72

File: archivist/archivist.py (hunk 6 above)
Comment (eccles) [reply archivist/archivist.py:349]: thanks

File: archivist/sboms.py (hunk 4 above)
Comment (eccles) [reply archivist/sboms.py:148]: I have added some - I will add more when doing withdraw/publish

**Pull request merged**